poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1086,38 +1215,51 @@ public PowerIQAPIClient createClient(FacilitySoftwareConfig poweriq) {
    }
 
    public List<Asset> filterAssetsBySource(String source, Set<String> assetIds) {
-      List<Asset> assets = new ArrayList<Asset>();
-      List<Asset> assetsFromPowerIQ = restClient.getAllAssetsBySource(source);
-      for (Asset asset : assetsFromPowerIQ) {
-         if (assetIds.contains(asset.getId())) {
-            assets.add(asset);
+      List<Asset> sensorAssets = new ArrayList<Asset>();
+      for(String assetId : assetIds) {
+         ResponseEntity<Asset> assetEntity = restClient.getAssetByID(assetId);
+         if(assetEntity == null) {
+            continue;
+         }
+         Asset asset = assetEntity.getBody();
+         if(!asset.getAssetSource().contains(source)) {
+            continue;
          }
+         sensorAssets.add(asset);
       }
-      return assets;
+      return sensorAssets;
    }
 
-   public void syncSensorRealtimeData(FacilitySoftwareConfig powerIQ) {
-      restClient.setServiceKey(serviceKeyConfig.getServiceKey());
-      List<Asset> allMappedAssets =
-            Arrays.asList(restClient.getMappedAsset(AssetCategory.Server).getBody());
-      if (allMappedAssets.isEmpty()) {
-         logger.info("No mapped server found. End sync RealTime data Job");
-         return;
-      }
-      List<RealTimeData> realTimeDatas = new ArrayList<RealTimeData>();
-      //get assetIds from asset's sensorsFromulars attribute
-      Set<String> assetIds = getAssetIdfromformular(allMappedAssets);
-      if (assetIds.isEmpty()) {
-         return;
-      }
-      //filter sensors
-      List<Asset> sensorFromPowerIQ = filterAssetsBySource(powerIQ.getId(), assetIds);
-      realTimeDatas = getSensorRealTimeData(powerIQ, sensorFromPowerIQ);
-      logger.info("Received new Sensor data, data item size is:" + realTimeDatas.size());
-      if (realTimeDatas.isEmpty()) {
-         return;
+   public Set<String> getSensorIdFromPdu(List<Asset> allMappedPdus, String assetSource){
+      Set<String> assetIds = new HashSet<String>();
+      for(Asset pdu : allMappedPdus) {
+         Map<String,String> justficationfields = pdu.getJustificationfields();
+         if(justficationfields.isEmpty()) {
+            continue;
+         }
+         String metricAndSensorIds =justficationfields.get(FlowgateConstant.SENSOR);
+         if(metricAndSensorIds == null) {
+            continue;
+         }
+         Map<String, String> metricAndSensorIdMap = null;
+         try {
+            metricAndSensorIdMap = getInfoMap(metricAndSensorIds);
+         } catch (IOException e) {
+            continue;
+         }
+         for(Map.Entry<String,String> map : metricAndSensorIdMap.entrySet()) {
+            String sensorIdAndSource = map.getValue();
+            String[] sensorInfos = sensorIdAndSource.split(FlowgateConstant.SPILIT_FLAG);
```

Comment:
I'm confused. it seems this is the formula map.Then how will it contain the asset source?

Reply:
it's useless,removed